internal/mempool/mempool.go

```diff
@@ -279,19 +279,14 @@ func (txmp *TxMempool) CheckTx(
 		return err
 	}
 
-	if txmp.recheckCursor != nil {
-		return errors.New("recheck cursor is non-nil")
-	}
```

Comment:
If its not too tedious, would we want to add an unit test for this so that it's caught in the future?